lib/Sema/TypeCheckGeneric.cpp

```diff
@@ -753,6 +753,28 @@ GenericSignatureRequest::evaluate(Evaluator &evaluator,
         inferenceSources.emplace_back(typeRepr, type);
       }
 
+      // Handle the thrown error type.
+      auto effectiveFunc = func ? func
+                                : subscr ? subscr->getEffectfulGetAccessor()
+                                : nullptr;
+      if (effectiveFunc) {
+        if (auto thrownTypeRepr = effectiveFunc->getThrownTypeRepr()) {
```

Comment:
Maybe mention here that this is for the `throws(E)` that's on the declaration itself

Reply:
I've done this in https://github.com/apple/swift/pull/69198